core/sdk/src/extn/client/extn_client.rs

```diff
@@ -285,6 +285,14 @@ impl ExtnClient {
                                 let mut ripple_context = self.ripple_context.write().unwrap();
                                 ripple_context.deep_copy(context);
                             }
+                            let is_ripple_context =
+                                RippleContext::is_ripple_context(&message.payload).is_none();
+
+                            if self.has_event_listener(&message.target.as_clear_string()) {
```

Comment:
Please add a negation here it should not proceed to send the ripple context if there are any listeners

Reply:
added the change please have a look